pkg/helm/manager.go

```diff
@@ -0,0 +1,106 @@
+package helm
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/replicatedhq/ekco/pkg/helm/charts/rookcephcluster"
+	"go.uber.org/zap"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/release"
+)
+
+type HelmManager struct {
+	ctx          context.Context
+	actionConfig *action.Configuration
+	logger       *zap.SugaredLogger
+}
+
+// NewHelmManager initializes a new HelmManager instance
+func NewHelmManager(ctx context.Context, namespace string, logger *zap.SugaredLogger) (*HelmManager, error) {
+
+	// Check if the namespace name is valid
+	if namespace == "" {
+		return nil, fmt.Errorf("error initializing Helm Manager: namespace must be set")
+	}
+
+	// Create a Helm client from the environment
+	env := cli.New()
+	env.SetNamespace(namespace)
+
+	// Create Helm Configuration
+	actConfig := action.Configuration{}
+	if err := actConfig.Init(env.RESTClientGetter(), namespace, "", logger.Debugf); err != nil {
+		return nil, fmt.Errorf("failed to initialize helm configuration: %w", err)
+	}
+
+	return &HelmManager{ctx: ctx, actionConfig: &actConfig, logger: logger}, nil
+}
+
+// InstallChartArchive installs a packaged Helm chart (tgz) to a namespace with given values and release name
+func (hm *HelmManager) InstallChartArchive(chartBytes io.Reader, values map[string]interface{}, releaseName string, namespace string) error {
+	chart, err := loader.LoadArchive(chartBytes)
+	if err != nil {
+		return err
+	}
+
+	relName := chart.Name()
+	if releaseName != "" {
+		relName = releaseName
+	}
+
+	// check if CephCluster CR was already installed
+	installed, err := hm.GetRelease(relName)
+	if err != nil {
+		return fmt.Errorf("failed to check if release %s is installed: %w", relName, err)
+	}
+
+	if installed != nil {
+		hm.logger.Infof("rook-ceph-cluster already installed with release name: %s", installed.Name)
+		return nil
+	}
+
+	act := action.NewInstall(hm.actionConfig)
+	act.Namespace = namespace
+	act.ReleaseName = relName
+	act.Wait = false
+
+	chartValues, err := rookcephcluster.ValuesMap()
+	if err != nil {
+		return fmt.Errorf("failed to get %s chart values: %w", chart.Name(), err)
+	}
+
+	// install chart
+	if _, err := act.RunWithContext(hm.ctx, chart, chartValues); err != nil {
```

Comment:
I think the problem here is the method.
I think you need to install/update with the chart values

Give a look in this implementation here for the manager : https://github.com/replicatedhq/brainstorming-ekco/pull/2/files#diff-8e7acbce7dcceea6f611e0a7e681d52cfc12fa9f4678fee8cebfa1e3bdbe5ec2

Reply:
As far as I can tell that's using the same helm API endpoint, just `Run` instead of `RunWithContext`, both of which go to the same actual install process

Reply:
For posterity: The problem with the helm chart not using the override values in `chartValues` was due to the yaml module we were using to unmarshal the `values.yaml` file. See https://github.com/replicatedhq/ekco/pull/121/commits/203259ab57345c492534ba94db549ac677de93ac